code/particle/util/ParticleProperties.cpp

```diff
@@ -37,21 +37,11 @@ void ParticleProperties::parse(bool nocreate) {
 int ParticleProperties::chooseBitmap()
 {
 	// failsafe check, though we should not have been able to get to this point
-	if (m_bitmap_list.empty())
-		return -1;
-
-	int bitmap_index;
-	int num_bitmaps = (int)m_bitmap_list.size();
-
-	if (num_bitmaps == 1) {
-		bitmap_index = m_bitmap_list[0];
-	} else if (num_bitmaps > 1) {
-		bitmap_index = m_bitmap_list[rand() % num_bitmaps];		
-	} else {
-		bitmap_index = -1;
-	}
+	Assertion(!m_bitmap_list.empty(), "No bitmaps found!");
+
+	size_t index = ::util::UniformRange<size_t>(0, m_bitmap_list.size() - 1).next();
```

Comment:
Creating a `RandomRange` is a potentially resource intensive operation so it would be better to only create this once when the bitmap list has been loaded and then just use that instance here.